thunder/torch/__init__.py

```diff
@@ -186,6 +186,92 @@ def is_cuda(a: TensorLike, /) -> bool:
 
 register_method("size", size)
 
+_torch_dtype_to_old_torch_typestring_map = {
+    torch.float32: "FloatTensor",
+    torch.float64: "DoubleTensor",
+    torch.float16: "HalfTensor",
+    torch.bfloat16: "BFloat16Tensor",
+    torch.uint8: "ByteTensor",
+    torch.int8: "CharTensor",
+    torch.int16: "ShortTensor",
+    torch.int32: "IntTensor",
+    torch.long: "LongTensor",
+    torch.bool: "BoolTensor",
+}
+
+_old_torch_typestring_to_torch_dtype_map = {v: k for k, v in _torch_dtype_to_old_torch_typestring_map.items()}
+
+
+def _device_and_dtype_to_old_torch_typestring(device: DeviceLike, dtype: dtypeLike):
+    torch_dtype = to_torch_dtype(dtype)
+    dtype_str = _torch_dtype_to_old_torch_typestring_map.get(torch_dtype)
+    devicetype_str: str = ""
+    if device.devicetype is not devices.DeviceType.CPU:
+        devicetype_str = f"{devices.devicetype_string(device.devicetype)}."
+    return f"torch.{devicetype_str}{dtype_str}"
+
+
+def _old_torch_typestring_to_devicetype_and_dtype(typestring: str) -> tuple[DeviceLike, dtypeLike]:
+
+    # Two cases:
+    #    - torch.DtypeTensor
+    #    - torch.device.DtypeTensor
+
+    _, *dev_and_dtype = typestring.split(".")
+
+    if len(dev_and_dtype) == 1:
+        (dtype_str,) = dev_and_dtype
+        return "cpu", _old_torch_typestring_to_torch_dtype_map[dtype_str]
+
+    if len(dev_and_dtype) == 2:
+        dtype_str, devicetype_str = dev_and_dtype
+        return dtype_str, _old_torch_typestring_to_torch_dtype_map[devicetype_str]
+
+    # Assertion error -- expected the string to split into one or two elements
+    utils.check(
+        False,
+        lambda: f"type(): dtype format does not match torch.Tensor dtype nor old torch typestring format",
+        exception_type=ValueError,
+    )
+
+
+@torchsymbol(torch.Tensor.type, is_method=True)
+def type(
+    a: TensorLike, /, dtype: None | str | dtypeLike = None, non_blocking: bool = False, **kwargs
```

Comment:
I don't think this wants to accept `**kwargs`?

Reply:
As per [PyTorch](https://pytorch.org/docs/stable/generated/torch.Tensor.type.html#torch.Tensor.type), some may use the keyword `async` and PyTorch implementation accepts that keyword although it is deprecated. Thought Thunder might want to do this as well.
Should I remove it as `async` is deprecated?

Reply:
Interesting! Yes — let's try not supporting it for now and see if anyone asks for `async`

Reply:
Sounds good!